src/_BirthDeath.pyx

```diff
@@ -59,26 +59,50 @@ cdef class BirthDeathModel:
 
 
     def __init__(self, number_of_sites, populations_number, number_of_susceptible_groups, seed, sampling_probability, memory_optimization):
-        self.rndm = RndmWrapper(seed=(seed, 0))
+        self.seed = RndmWrapper(seed=(seed, 0))
+        self.internal_seed = int(floor(self.seed.uniform()*1000000))
+        self.rndm = RndmWrapper(seed=(self.internal_seed, 0))
 
         self.first_simulation = False
         self.sampling_probability = sampling_probability
-        if memory_optimization == None:
-            self.memory_optimization = False
-        else:
-            self.memory_optimization = True
 
-        self.seed = seed
         self.sites = number_of_sites
         self.hapNum = 4**self.sites
-        self.currentHapNum = 0
-        if self.memory_optimization == True:
+        self.susNum = number_of_susceptible_groups
+        self.popNum = populations_number
+
+        #Memory optimization
+        if isinstance(memory_optimization, list) and len(memory_optimization) == 2:
```

Comment:
is this polymorphism on `memory_optimization` really needed? A list of two elements (not a tuple! just a list, ugh) or an integer or a boolean. You're clearly trying to shove something into it, what is it?